### PR TITLE
Feature/docker and k8s

### PR DESCRIPTION
**Pull request by ajax-makarytskyi-o**
Added docker, docker compose and kubernetes to project.

### CODE REVIEW

File: k8s/config-map.yaml
Comment (ajax-torbin-v) [inline k8s/config-map.yaml:1]: Naming is kinda confusing 
Better rename to **mongo-config-map**

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    environment:
+      database: ${database}
+      username: ${mongo_username}
+      password: ${mongo_password}
+      authentication_database: ${authentication_database}
+      mongo_host: ${mongo_host}
+      mongo_port: ${mongo_port}
+
+  mongo:
+    image: mongo:latest
```
Comment (ajax-malov-n) [inline docker-compose.yml]: Specify image version

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    container_name: mongo
+    restart: always
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: ${mongo_username}
```
Comment (ajax-malov-n) [inline docker-compose.yml]: Use upper case for constants

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -4,17 +4,7 @@ spring:
...
-      database: cars
-      username: mongoadmin
-      password: mongopassword
-      authentication-database: admin
-
-  autoconfigure:
-    exclude:
-      - org.springframework.boot.autoconfigure.mongo.MongoAutoConfiguration
-      - org.springframework.boot.autoconfigure.data.mongo.MongoDataAutoConfiguration
+      uri: mongodb://${username}:${password}@${mongo_host}:${mongo_port}/${database}?authSource=${authentication_database}
```
Comment (ajax-malov-n) [inline src/main/resources/application.yml]:
```suggestion
      uri: mongodb://${username}:${password}@${mongo_host}:${mongo_port}/${database}?authSource=admin
```

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    depends_on:
+      - mongo
+    ports:
+      - "8080:8080"
+    environment:
+      DATABASE: ${DATABASE}
+      USERNAME: ${MONGO_USERNAME}
+      PASSWORD: ${MONGO_PASSWORD}
+      MONGO_HOST: ${MONGO_HOST}
+      MONGO_PORT: ${MONGO_PORT}
```
Comment (ajax-gnatiuk-s) [inline docker-compose.yml]: you have .env file, so it's good idea to reuse it here as well

File: k8s/start.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
+minikube start
+cd ..
+./gradlew build
+eval $(minikube docker-env)
+docker build -t application .
+cd k8s
```
Comment (ajax-gnatiuk-s) [inline k8s/start.sh]: don't use `cd` command in the script

File: k8s/start.sh [hunk 6]
```diff
@@ -0,0 +1,21 @@
+minikube start
+cd ..
+./gradlew build
+eval $(minikube docker-env)
```
Comment (ajax-gnatiuk-s) [inline k8s/start.sh]: why do we need it?

File: k8s/mongo.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
+apiVersion: apps/v1
+kind: Deployment
```
Comment (ajax-gnatiuk-s) [inline k8s/mongo.yaml]: why not `StatefulSet`?

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,7 @@
+FROM openjdk:21-jdk
+
+WORKDIR /app
+
+COPY ./build/libs/*.jar ./rentcar.jar
```
Comment (ajax-gnatiuk-s) [inline Dockerfile]: u have few `*.jar` files in the build dir. What would be copied?

File: k8s/start.sh (hunk 6 above)
Comment (ajax-makarytskyi-o) [reply k8s/start.sh]: to rebuild project and use newer version if there were changes

File: Dockerfile (hunk 8 above)
Comment (ajax-makarytskyi-o) [reply Dockerfile]: the first one

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,7 @@
+FROM openjdk:21-jdk
+
+WORKDIR /app
+
+COPY ./build/libs/rentcar-0.0.1-SNAPSHOT.jar ./rentcar.jar
```
Comment (ajax-gnatiuk-s) [inline Dockerfile]:
```suggestion
COPY ./build/libs/rentcar-*.jar ./rentcar.jar
```

Please, copy by pattern. Sometimes we update the apps version

**Review by ajax-gnatiuk-s — APPROVED** [review]

**Pull request merged**